src/main/java/com/rabbitmq/client/impl/recovery/AutorecoveringChannel.java

```diff
@@ -900,7 +900,11 @@ private void recordConsumer(String result,
         this.connection.recordConsumer(result, consumer);
     }
 
-    private void deleteRecordedConsumer(String consumerTag) {
+    /**
+     * Delete the recorded consumer from this channel and accompanying connection
+     * @param consumerTag consumer tag to delete
+     */
+    public void deleteRecordedConsumer(String consumerTag) {
```

Comment:
useful in scenarios where a consumer is canceled by the server, and we want to remove it from the client side recovery